src/wallet/wallet.cpp

```diff
@@ -2729,7 +2762,14 @@ bool CWallet::CreateTransaction(const std::vector<CRecipient>& vecSend, CTransac
                 if (pick_new_inputs) {
                     nValueIn = 0;
                     setCoins.clear();
-                    coin_selection_params.change_spend_size = CalculateMaximumSignedInputSize(change_prototype_txout, this);
+                    int change_spend_size = CalculateMaximumSignedInputSize(change_prototype_txout, this);
+                    // If the wallet doesn't know how to sign change output, assume p2sh-p2wpkh
+                    // as lower-bound to allow BnB to do it's thing
+                    if (change_spend_size == -1) {
+                        coin_selection_params.change_spend_size = (size_t)CalculateNestedKeyhashInputSize(false);
```

Comment:
We should handle the case of `-1` here too, or if `CalculateNestedKeyhashInputSize` is guaranteed to never return negative values it should return an unsigned type? :-)

Reply:
Changed the signature to return `size_t` directly since there's no reason these things need to be signed...

Reply:
Then the cast can be removed? :-)